### ISSUE TITLE
AutoLaborManager crashes if Dig designation is placed above unallocated map_block

### ISSUE THREAD

**Issue by espresso4** [DFHack/dfhack#1240]
Every time I use digtype on gold it crashes either immediately or after I use it again on another gold nugget. It only happens with gold.

**Comment by espresso4** [comment]
Actually, it crashed when I tried it on tetrahedrite. It crashes each time I try to use it on that ore.

Odd, I was able to use it on various iron ores and platinum without issue.

**Comment by quietust** [comment]
What version of DFHack+DF, and on what platform+architecture?

**Comment by espresso4** [comment]
DF 44.07 and DFHack 44.07-Alpha1

No other mods

Windows 10 64 bit

**Comment by quietust** [comment]
I've just tested digtype with DFHack 0.44.07-alpha1 with the 64-bit Windows build and was unable to observe any crashes. However, I **did** observe it placing Dig designations on top of raw adamantine, semi-molten rock, and _slade_ where mineral inclusions likely would have appeared (had they not been overwritten by deep special tubes, the magma core, and the underworld).

**Comment by espresso4** [comment]
Hmmm, I can replicate this issue quite easily.

Maybe I should add one last detail I thought was not important.

I binded the “digtype” command to Ctrl-Shift-Alt-V so I use those 4 buttons each time rather than typing it. This is in the dfhack init file where you edit shortcuts. I checked that no other command used this shortcut.

Can you give me some tips on what I should try to verify this as a bug? Should I try to replicate it on another computer? Try a fresh install (even though mine is just DF and DFhack). Let me know.

**Comment by quietust** [comment]
You need to upload a copy of your fortress to DFFD or Dropbox (or some other file sharing site) so one of us can see what's causing it to crash - it's quite likely that the crash is related to the geology of your fortress's embark location.

**Comment by espresso4** [comment]
http://dffd.bay12games.com/file.php?id=13584

Here is my save folder. I did create the world using advanced world gen. It may have something to do with that.

**Comment by quietust** [comment]
An important detail you neglected to mention: the game does not crash until you **unpause** it.

The crash seems to be happening in AutoLaborManager::count_map_designations(void), on the following line:

```
                        df::coord p = bl->map_pos;
                        df::coord c(p.x, p.y, p.z - 1);
>                       if (Maps::getTileDesignation(c)->bits.hidden)
                            continue;
```
In this specific case, the coordinate "c" is (48,0,3), which is pointing at an _unallocated_ map block.

**Comment by espresso4** [comment]
Ah yes that is true. If the game is paused it does not crash after using digtype. Only once I unpause does it crash.

I'm glad I was able to help. First time posting here.

**Comment by ab9rf** [comment]
It looks to me that, irrespective of any issues with digtype, I need to add a check in labormanager for the possibility of a mapsquare on the lowest level of the map being designated.

**Comment by lethosor** [comment]
Maps::isTileVisible should work in this case.

**Comment by lethosor** [comment]
@espresso4 just to confirm, if you disable labormanager, does the issue go away?

In general, mentioning what other plugins/tools you have enabled is a good idea. Labormanager isn't enabled in the default config files, so most people wouldn't think of enabling it to test a digtype bug. You can run `install-info` to get a full list.

**Comment by ab9rf** [comment]
@lethosor just to note, @PeridexisErrant made labormanager on by default in his starter pack a while back, so people using his starter pack will have it enabled by default.

**Comment by lethosor** [comment]
I remember him changing labormanager to be the thing that "automatic labor management" (or whatever it's called in PyLNP) controls, but I doubt it's enabled by default. And in any case, I don't use that pack, so I don't know what users of it have enabled.

**Comment by espresso4** [comment]
@lethosor I enabled labormanager myself. It was not on by default as I installed each mod separately. First bug report ever so now I know all the details I should include.

And correct, the game does not crash if I disable labormanager. It only crashes while labormanager is on and I use digtype on certain veins.

@ab9rf The latest releases of @PeridexisErrant don't have autolabor nor labormanager enabled automatically.

Thanks for all the awesome work!
